### PR TITLE
cabal-docspec: Use OS-appropriate newline strings

### LINKED ISSUE #56: cabal-docspec fails to run on Windows
To reproduce:

```
$ git clone https://github.com/phadej/comonad && cd comonad
$ git reset --hard 00621cac336377f71bf9a0be13744b2836cfc733
$ cabal build all --write-ghc-environment-files=always
$ .../cabal-docspec
[   0.00005] ghc.info: ghc
[   0.00223] process.0.start: cwd=C:\Users\winferno\AppData\Local\Temp ghc --info
[   0.05379] docspec.component: comonad-5.0.7 lib
[   0.05841] docspec.phase1: Control.Comonad: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad.hs
[   0.06579] docspec.phase1: Control.Comonad.Env: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Env.hs
[   0.07031] docspec.phase1: Control.Comonad.Env.Class: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Env\Class.hs
[   0.07410] docspec.phase1: Control.Comonad.Hoist.Class: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Hoist\Class.hs
[   0.07744] docspec.phase1: Control.Comonad.Identity: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Identity.hs
[   0.08187] docspec.phase1: Control.Comonad.Store: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Store.hs
[   0.08679] docspec.phase1: Control.Comonad.Store.Class: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Store\Class.hs
[   0.09265] docspec.phase1: Control.Comonad.Traced: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Traced.hs
[   0.09710] docspec.phase1: Control.Comonad.Traced.Class: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Traced\Class.hs
[   0.10272] docspec.phase1: Control.Comonad.Trans.Class: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Trans\Class.hs
[   0.10801] docspec.phase1: Control.Comonad.Trans.Env: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Trans\Env.hs
[   0.11432] docspec.phase1: Control.Comonad.Trans.Identity: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Trans\Identity.hs
[   0.11884] docspec.phase1: Control.Comonad.Trans.Store: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Trans\Store.hs
[   0.12624] docspec.phase1: Control.Comonad.Trans.Traced: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Control\Comonad\Trans\Traced.hs
[   0.13197] docspec.phase1: Data.Functor.Composition: C:\Users\winferno\Documents\Hacking\Haskell\comonad\.\src\Data\Functor\Composition.hs
[   3.17884] debug: stderr:
Some flags have not been recognized: prompt2,

[   3.18126] debug: stdout:
GHCi, version 8.10.3: https://www.haskell.org/ghc/  :? for help
Prelude> "5d6fbp77nbdu2k345ud3qzexbf5lehsrqjb2q4vj7ntbgr7zf"

[   3.18529] peu.die: Timeout while starting GHCi
proci: canceling
proci: terminating
proci: pid Just 7888
proci: waiting for termination
```

I can only reproduce this on Windows, not on Linux.

### PR DESCRIPTION
**Pull request by RyanGlScott**
Previously, `cabal-docspec` looked specifically for the character `\n`, which doesn't work on Windows due to its use of `\r\n`. To account for this, this patch introduces a more portable way to look for newline strings.

Fixes #56.

### CODE REVIEW

File: cabal-docspec/src/CabalDocspec/GHCi.hs [hunk 1]
```diff
@@ -104,10 +105,10 @@ waitGhci _tracer (GHCi iph mt) mitVar microsecs = do
 
     -- send separator
     separator <- show <$> genString
-    Proci.sendTo iph $ fromString $ separator ++ " :: " ++ mt' ++ "\n"
+    Proci.sendTo iph $ fromString $ separator ++ " :: " ++ mt' ++ newlineStr
```
Comment (phadej) [inline cabal-docspec/src/CabalDocspec/GHCi.hs:108]: Have you tested that `\r\n` doesn't cause any weirdness when *sent to*?

Comment (RyanGlScott) [reply cabal-docspec/src/CabalDocspec/GHCi.hs:108]: I can successfully run the doctests for `comonad` and `lens` with this patch. Is there some other way you'd like for me to test this?

Comment (phadej) [reply cabal-docspec/src/CabalDocspec/GHCi.hs:108]: That's fine with me.

**Comment by phadej** [comment]
FYI, `cabal run cabal-docspec` "self-checks" too. Even here isn't many doctests, it's often enough.

**Pull request merged**